bayespy/inference/vmp/vmp.py

```diff
@@ -144,6 +144,11 @@ def update(self, *nodes, repeat=1, plot=False, tol=None, verbose=True):
         else:
             plot_nodes = [self[x] for x in plot]
 
+        # Make certain that at least one of the nodes in the update list
+        # has been observed
+        if ~np.any([n.observed for n in nodes]):
```

Comment:
Yep, this looks nice, thanks! But this doesn't actually work in all cases, because `observed` can be a boolean *array*, where each element tells whether the corresponding element in the node has been observed or not. Yep, this information isn't probably documented anywhere.. Anyway, it enables observing nodes partially. Changing the `if` statement as follows should fix:
```
if all(~np.any(n.observed) for n in nodes):
```
But I can merge this pull request and make this change myself. Thanks!